src/conversation_engine/engine.py

```diff
@@ -374,47 +402,53 @@ def _handle_command(self, text):
             self._start_new_conversation()
 
     def _handle_additional_info(self, text):
-        """Parse text into additional info according to grammar & target received from action_server result"""
+        """
+        Parse text into additional info according to grammar & target received from action_server result
+        """
         rospy.loginfo("_handle_additional_info('{}')".format(text))
 
         words = text.strip().split(" ")
-        additional_semantics = self._parser.parse(self._state.target, words, debug=True)
-
-        rospy.loginfo("additional_semantics: {}".format(additional_semantics))
-        sem_str = json.dumps(additional_semantics)
-        sem_dict = yaml.load(sem_str)
-        rospy.logdebug("parsed: {}".format(sem_dict))
-
-        if additional_semantics:
-            try:
-                rospy.loginfo("Additional_semantics: {}".format(additional_semantics))
-                self._state.update_semantics(sem_dict, self._state.missing_field)
-                self._say_to_user(random.choice(["OK, that helps!",
-                                                 "Allright, thanks!"]))
-
-                self._action_client.send_async_task(str(self._state.current_semantics),
-                                                    done_cb=self._done_cb,
-                                                    feedback_cb=self._feedback_cb)
-                rospy.loginfo("Updated task sent: {}".format(self._state.current_semantics))
-
-                self._state.wait_for_robot()
-            except (KeyError, IndexError) as e:
-                rospy.logerr("Could not update semantics: {}".format(e))
-                self._say_to_user(
-                    random.choice(["Something went terribly wrong, can we try a new command?",
-                                   "I didn't understand that, what do you want me to do?",
-                                   "What would you like me to do? Could you please rephrase you command?"]))
-                self._stop()
-        else:
+        try:
+            additional_semantics = self._parser.parse_raw(self._state.target, words, debug=True)
```

Comment:
The variable `additional_semantics` is now created in the scope of this `try/except` and thus does not exist outside that scope if the `try` fails and thus some of the statements below will fail because the variable they use doesn't exist. 

If `additional_semantics` is initialized to `{}`, we should be fine

Reply:
The 'except' contains a return

Reply:
Aha, there is a `return` in the `except` clause so it's not possible to continue when it false